crates/cxx-qt-lib/src/gui/qimage.rs

```diff
@@ -11,12 +11,100 @@ mod ffi {
+        #[rust_name = "is_gray_scale"]
+        fn isGrayscale(self: &QImage) -> bool;
+
+        /// Returns true if the image has a format that respects the alpha channel, otherwise returns false.
+        #[rust_name = "has_alpha_channel"]
+        fn hasAlphaChannel(self: &QImage) -> bool;
+
+        /// Returns a sub-area of the image as a new image.
+        fn copy(self: &QImage, rect: &QRect) -> QImage;
+
+        /// Returns the size of the color table for the image.
+        #[rust_name = "color_count"]
+        fn colorCount(self: &QImage) -> i32;
+
+        /// Resizes the color table to contain colorCount entries.
+        #[rust_name = "set_color_count"]
+        fn setColorCount(self: &mut QImage, colorCount: i32);
+
+        /// Returns the depth of the image.
+        fn depth(self: &QImage) -> i32;
+
+        /// Fills the entire image with the given color.
+        fn fill(self: &mut QImage, color: &QColor);
+
+        /// Returns the height of the image.
+        fn height(self: &QImage) -> i32;
+
+        /// Returns the width of the image.
+        fn width(self: &QImage) -> i32;
+
+        /// Returns the size of the image.
+        fn size(self: &QImage) -> QSize;
+
+        /// Returns the enclosing rectangle (0, 0, width(), height()) of the image.
+        fn rect(self: &QImage) -> QRect;
+
+        /// Returns the color of the pixel at coordinates (x, y) as a QColor.
+        #[rust_name = "pixel_color"]
+        fn pixelColor(self: &QImage, x: i32, y: i32) -> QColor;
+
+        /// Sets the pixel color at (x, y) to color.
+        #[rust_name = "set_pixel_color"]
+        fn setPixelColor(self: &mut QImage, x: i32, y: i32, color: &QColor);
+
+        /// Returns the number of pixels that fit horizontally in a physical meter.
+        #[rust_name = "dots_per_meterx"]
+        fn dotsPerMeterX(self: &QImage) -> i32;
+
+        /// Returns the number of pixels that fit vertically in a physical meter.
+        #[rust_name = "dots_per_metery"]
+        fn dotsPerMeterY(self: &QImage) -> i32;
+
+        /// Returns true if pos is a valid coordinate pair within the image.
+        fn valid(self: &QImage, x: i32, y: i32) -> bool;
+
+        /// Returns the number of pixels by which the image is intended to be offset by when positioning relative to other images.
+        fn offset(self: &QImage) -> QPoint;
+
+        /// Sets the number of pixels by which the image is intended to be offset by when positioning relative to other images, to offset.
+        #[rust_name = "set_offset"]
+        fn setOffset(self: &mut QImage, point: &QPoint);
+
+        /// Returns the pixel index at (x, y).
+        #[rust_name = "pixel_index"]
+        fn pixelIndex(self: &QImage, x: i32, y: i32) -> i32;
+
+        /// Mirrors of the image in the horizontal and/or the vertical direction depending on whether horizontal and vertical are set to true or false.
+        fn mirror(self: &mut QImage, horizontal: bool, vertical: bool);
```

Comment:
mirror doesn't exist in Qt 5.

We do have a cfg option for the Qt major version we use elsewhere

```rust
#[cfg(qt_version_major = "5")]
```

However I'm not sure if this works within a CXX bridge.

So what you might need to do is create a second bridge for the Qt 6+ features

```rust
#[cfg(not(qt_version_major = "5"))]
#[cxx::bridge]
mod ffi_qt_6 {
   ...  // Qt 6 specific methods
}
```